src/utils/contracts/AdapterInvestLimitBase.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: gpl-3.0
+
+pragma solidity >=0.6.0 <0.9.0;
+
+//  libraries
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+
+// helper contracts
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import { AdapterModifiersBase } from "./AdapterModifiersBase.sol";
+
+import "../../interfaces/defiAdapters/contracts/IAdapterInvestLimit.sol";
+
+abstract contract AdapterInvestLimitBase is IAdapterInvestLimit, AdapterModifiersBase {
+    using Address for address;
+    using SafeMath for uint256;
+
+    /** @notice max deposit value datatypes */
+    MaxExposure public maxDepositProtocolMode;
+
+    /** @notice max deposit's protocol value in percentage */
+    uint256 public maxDepositProtocolPct; // basis points
+
+    /** @notice  Maps liquidityPool to max deposit value in percentage */
+    mapping(address => uint256) public maxDepositPoolPct; // basis points
+
+    /** @notice  Maps liquidityPool to max deposit value in absolute value for a specific token */
+    mapping(address => mapping(address => uint256)) public maxDepositAmount;
+
+    constructor() public {
+        maxDepositProtocolPct = uint256(10000); // 100% (basis points)
+        maxDepositProtocolMode = MaxExposure.Pct;
+    }
+
+    /**
+     * @inheritdoc IAdapterInvestLimit
+     */
+    function setMaxDepositPoolPct(address _liquidityPool, uint256 _maxDepositPoolPct)
+        external
+        override
+        onlyRiskOperator
+    {
+        maxDepositPoolPct[_liquidityPool] = _maxDepositPoolPct;
+        emit LogMaxDepositPoolPct(maxDepositPoolPct[_liquidityPool], msg.sender);
+    }
```

Comment:
Is there any situation in which the `RiskOperator` needs to set the `maxDepositPoolPct[_liquidityPool]` to zero? because it is not going to work when calling `getMaxDepositAmountByPct()`